QuestionLibrary/admin.py

```diff
@@ -59,6 +62,103 @@ def load_selected_records_action(modeladmin, request, queryset):
+
+    def get_readonly_fields(self, request, obj=None):
+        return list(super().get_fields(request, obj))
+
+    def has_add_permission(self, request, obj=None):
+        return False
+
+
+class EPAResponseForm(ModelForm):
+    def __init__(self, *args, **kwargs):
+        super(EPAResponseForm, self).__init__(*args, **kwargs)
+        # turn the layer id fields into dropdowns populated with the layers in base service config JSON
+        # field labels as defined in the model would be overwritten, so we copy those to the new dropdown fields
+        if self.instance.map_service_config is not None:
+            self.fields['system_layer_id'] = forms.ChoiceField(choices=self.instance.get_layers_as_choices(),
+                                                               label=self.fields['system_layer_id'].label)
+            self.fields['facility_layer_id'] = forms.ChoiceField(choices=self.instance.get_layers_as_choices(),
+                                                                 label=self.fields['facility_layer_id'].label)
+            self.fields['assessment_table_id'] = forms.ChoiceField(choices=self.instance.get_tables_as_choices(),
+                                                                   label=self.fields['assessment_table_id'].label)
+        # else:
+        #     self.fields['system_layer_id'].widget = forms.HiddenInput()
+        #     self.fields['facility_layer_id'].widget = forms.HiddenInput()
+        #     self.fields['assessment_table_id'].widget = forms.HiddenInput()
+
+    #  todo add button to grab attributes from base data and post to survey123 service. this might not be the best place for this. need to think this through and ask karl
+
+
+@admin.register(EPAResponse)
+class EPAResponseAdmin(admin.ModelAdmin):
+    inlines = [SurveyInline]
+    actions = [disable_epa_response_action, enable_epa_response_action]
+    readonly_fields = ['_status', '_map_service_config']
+    exclude = ['disabled_date', 'map_service_config']
+    list_display = ['name', '_status', '_map_service_config']
+    list_filter = ['disabled_date']
+    search_fields = ['name']
+    form = EPAResponseForm
+
+    def _status(self, obj=None):
+        if obj.disabled_date is None:
+            return 'Active'
+        else:
+            return f'Disabled {obj.disabled_date.strftime("%Y-%m-%d")}'
+
+    def _map_service_config(self, obj=None):
+        if obj.map_service_config is None:
+            return '-'
+        else:
+            config = json.loads(obj.map_service_config)
+            if 'error' in config:
+                return 'Error getting configuration'
+            if 'layers' in config:
+                return 'Configuration Downloaded'
+
+    def save_model(self, request, obj, form, change):
+        if 'map_service_url' in form.changed_data:
+            obj.get_map_service(request.user)
+        super().save_model(request, obj, form, change)
+
+    def get_search_results(self, request, queryset, search_term):
```

Comment:
If not dealing fk/relationships I think you can use limit_choices_to on the model fk field.